src/main/java/de/tum/cit/aet/artemis/communication/service/notifications/MailService.java

```diff
@@ -207,10 +207,18 @@ public void sendSuccessfulDataExportsEmailToAdmin(User admin, Set<DataExport> da
      * @return the modified subject of the email
      */
     private String createAnnouncementText(Object notificationSubject, Locale locale) {
+        // Fallback course title
+        String unknownCourseEn = "(Unknown Course)";
+        String unknownCourseDe = "(Unbekannter Kurs)";
+        String unknownCourseText = locale.toString().equals("en") ? unknownCourseEn : unknownCourseDe;
+
         // Translation that can not be done via i18n Resource Bundle (for Thymeleaf) but has to be set in this service via Java
         String newAnnouncementString = locale.toString().equals("en") ? "New announcement \"%s\" in course \"%s\"" : "Neue Ankündigung \"%s\" im Kurs \"%s\"";
-        String postTitle = ((Post) notificationSubject).getTitle();
-        String courseTitle = ((Post) notificationSubject).getConversation().getCourse().getTitle();
+
+        Post post = (Post) notificationSubject;
```

Comment:
Without checking that this is a post, it should not be casted. btw. why is the type "Object"?